### PR TITLE
fix: issues with not found errors

### PR DESCRIPTION
**Pull request by rdumont**
This PR fixes very common issues #1 and #2.

### CODE REVIEW

File: lib/release.js [hunk 1]
```diff
@@ -13,21 +13,18 @@ module.exports.createRelease = async function createRelease ({
...
-    head: targetTag,
-    ...repo
-  })
+/**
+ * @param {import('./github').GitHubService} gh
+ * @param {string} targetTag
+ */
+async function createReleaseWithService (gh, targetTag) {
+  const previousTag = await gh.getPreviousTag(targetTag)
+  const commits = await gh.repo(previousTag, targetTag)
```
Comment (weslleyss12) [inline lib/release.js:27]: Here you're calling the `repo` method from your `GitHubService` to get your commits, but this method only exists inside `Octokit`, so you could call it this way:
```suggestion
  const commits = await gh.gh.repo(previousTag, targetTag)
```
or even better use the `compareCommits` method you've implemented to retrieve the commits.
```suggestion
  const commits = await gh.compareCommits(previousTag, targetTag)
```

Comment (rdumont) [reply lib/release.js:27]: Totally a bug, the intention was to call `compareCommits` anyway.

**Comment by rdumont** [comment]
> `NIT` Don't forget to update the PR title to use a conventional commit pattern.

Not a nit, completely relevant! Thanks.

File: lib/release.js (hunk 1 above)
Comment (rdumont) [reply lib/release.js]: Fixed on https://github.com/escaletech/circleci-gh-conventional-release/pull/5/commits/0db0c447c7109e274f2970cbe0678953f4091bf2.

I'll follow up on this PR with another one to improve testing.

**Pull request merged**